packages/as-sha256/assembly/index.ts

```diff
@@ -158,25 +159,17 @@ function hashBlocks(wPtr: usize, mPtr: usize): void {
   g = H6;
   h = H7;
 
-  // Load message blocks into first 16 expanded message blocks
-  for (i = 0; i < 16; i++) {
-    store32(wPtr, i,
-      load32be(mPtr, i)
-    );
-  }
-  // Expand message blocks 17-64
-  for (i = 16; i < 64; i++) {
-    store32(wPtr, i,
-      SIG1(load32(wPtr, i - 2)) +
-      load32(wPtr, i - 7) +
-      SIG0(load32(wPtr, i - 15)) +
-      load32(wPtr, i - 16)
-    );
-  }
+  // 16 first u32 of expanded message block are same as message block
+  // rest of the 48 u32 are computed from the first 16 u32
+  // instead of computing expanded messsage blocks in a separate for loop,
+  // chain it to the below for loop to improve performance
 
   // Apply SHA256 compression function on expanded message blocks
   for (i = 0; i < 64; i++) {
-    t1 = h + EP1(e) + CH(e, f, g) + load32(kPtr, i) + load32(wPtr, i);
+    tmpW = i < 16 ? load32be(mPtr, i) : SIG1(load32(wPtr, i - 2)) + load32(wPtr, i - 7) +
```

Comment:
It looks like `tmpW` could be declared inside the loop? Like `const tmpW = ..`. It appears to be only assigned once per iteration and not updated.

Reply:
Or is it some assemblyscript specifics?

Reply:
we don't want to allocate variables there to avoid `gc`, it's designed like that for all other places as well.
yes, it's specific to assemblyscript